executor/sort.go

```diff
@@ -47,21 +47,29 @@ type SortExec struct {
 	// keyCmpFuncs is used to compare each ByItem.
 	keyCmpFuncs []chunk.CompareFunc
 	// rowChunks is the chunks to store row values.
-	rowChunks *chunk.List
+	rowChunks *chunk.RowContainer
 	// rowPointer store the chunk index and row index for each row.
 	rowPtrs []chunk.RowPtr
 
 	memTracker  *memory.Tracker
 	diskTracker *disk.Tracker
 
-	// rowChunksInDisk is the chunks to store row values in disk.
-	rowChunksInDisk *chunk.ListInDisk
-	// rowPtrsInDisk store the disk-chunk index and row index for each row.
-	rowPtrsInDisk []chunk.RowPtr
-	// partitionList is the chunks to store row values in disk for partitions.
-	partitionList []*chunk.ListInDisk
-	// partitionRowPtrs store the disk-chunk index and row index for each row for partitions.
+	// partitionList is the chunks to store row values for partitions.
+	partitionList []*chunk.RowContainer
+	// partitionRowPtrs store the sorted RowPtrs for each row for partitions.
 	partitionRowPtrs [][]chunk.RowPtr
+
+	// sortRows is used to maintain a heap.
+	sortRows []chunk.Row
+	// sortRowsIndex store the partition index for each row.
+	sortRowsIndex []int
```

Comment:
How about removing these two fields here and putting them into `multiWayMerge` and then let `SortExec` reference a `multiWayMerge` pointer?

Reply:
I think your idea is good. I will address comment.